kinto_amo/__init__.py

```diff
@@ -15,14 +16,33 @@ def includeme(config):
     # Parse resources
     settings = config.get_settings()
     resources = OrderedDict()
-    resources['addons'] = parse_resource(
-        settings.get('amo.addons', DEFAULT_ADDONS))
-    resources['plugins'] = parse_resource(
-        settings.get('amo.plugins', DEFAULT_PLUGINS))
-    resources['gfx'] = parse_resource(
-        settings.get('amo.gfx', DEFAULT_GFX))
-    resources['certificates'] = parse_resource(
-        settings.get('amo.certificates', DEFAULT_CERTIFICATES))
+
+    # Configure default settings
+    resources['blocklist'] = OrderedDict()
+    resources['blocklist']['addons'] = parse_resource(DEFAULT_ADDONS)
+    resources['blocklist']['plugins'] = parse_resource(DEFAULT_PLUGINS)
+    resources['blocklist']['gfx'] = parse_resource(DEFAULT_GFX)
+    resources['blocklist']['certificates'] = parse_resource(DEFAULT_CERTIFICATES)
+
+    # Read blocklist settings
+    for settings_key, settings_value in settings.items():
+        if not settings_key.startswith('amo.'):
+            continue
+        parts = settings_key.split('.')
+
+        if len(parts) == 2:  # amo.addons
+            prefix = 'blocklist'
+            _, blocklist = parts
+        else:  # amo.preview.addons
+            _, prefix, blocklist = parts
```

Comment:
Then it will fail again if blocklist contains a . on line 35.
I don't think we support . in the blocklist name.